pkg/util/services/basic_service.go

```diff
@@ -278,6 +278,9 @@ func (b *BasicService) awaitState(ctx context.Context, expectedState State, ch c
 
 		// if service has failed, include failure case in the returned error.
 		if failure := b.FailureCase(); failure != nil {
+			if failure == ErrStopCortex {
```

Comment:
This doesn't make sense. The services package should be agnostic of Cortex, but looks like that might not be the case with this.